mParticle-Rokt/MPKitRokt.m

```diff
@@ -63,6 +72,85 @@ - (void)start {
+///
+/// \param onUnLoad Function to execute right after widget is unloaded, there is no widget or there is an exception
+///
+/// \param onShouldShowLoadingIndicator Function to execute when the loading indicator should be shown
+///
+/// \param onShouldHideLoadingIndicator Function to execute when the loading indicator should be hidden
+///
+/// \param onEmbeddedSizeChange Function to execute when size of embeddedView change, the first item is selected
+/// Placement and second item is widget height
+///
+- (MPKitExecStatus *)executeWithViewName:(NSString * _Nullable)viewName
+                              attributes:(NSDictionary<NSString *, NSString *> * _Nonnull)attributes
+                              placements:(NSDictionary<NSString *, RoktEmbeddedView *> * _Nullable)placements
+                                  onLoad:(void (^ _Nullable)(void))onLoad
+                                onUnLoad:(void (^ _Nullable)(void))onUnLoad
+            onShouldShowLoadingIndicator:(void (^ _Nullable)(void))onShouldShowLoadingIndicator
+            onShouldHideLoadingIndicator:(void (^ _Nullable)(void))onShouldHideLoadingIndicator
+                    onEmbeddedSizeChange:(void (^ _Nullable)(NSString * _Nonnull, CGFloat))onEmbeddedSizeChange
+                            filteredUser:(FilteredMParticleUser * _Nonnull)filteredUser {
+    NSDictionary<NSString *, NSString *> *mpAttributes = [filteredUser.userAttributes transformValuesToString];
+    NSMutableDictionary<NSString *, NSString *> *finalAtt = [[NSMutableDictionary alloc] init];
+    [finalAtt addEntriesFromDictionary:@{@"mpid": filteredUser.userId.stringValue}];
+    [finalAtt addEntriesFromDictionary:mpAttributes];
+    
+    [Rokt executeWithViewName:viewName
+                   attributes:finalAtt
+                   placements:[self confirmPlacements:placements]
+                       onLoad:onLoad
+                     onUnLoad:onUnLoad
+ onShouldShowLoadingIndicator:onShouldShowLoadingIndicator
+ onShouldHideLoadingIndicator:onShouldHideLoadingIndicator
+         onEmbeddedSizeChange:onEmbeddedSizeChange
+    ];
+    
+    return [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
+}
+
+-(NSDictionary<NSString *, RoktEmbeddedView *> * _Nullable) confirmPlacements:(NSDictionary<NSString *, RoktEmbeddedView *> * _Nullable)placements {
+    NSMutableDictionary <NSString *, RoktEmbeddedView *> *safePlacements = [NSMutableDictionary dictionary];
+    
+    for (NSString* key in placements) {
+        id value = [placements objectForKey:key];
+        
+        if ([value isKindOfClass:RoktEmbeddedView.class]) {
+            [safePlacements setObject:value forKey:key];
+        }
+    }
+    
+    return safePlacements;
+}
+
+- (NSDictionary<NSString *, NSString *> *) filteredUserAttributes:(NSDictionary<NSString *, NSString *> * _Nonnull)attributes kitConfiguration:(MPKitConfiguration *)kitConfiguration {
```

Comment:
Is there a plan to add tests for this?

Reply:
Added tests but since the core sdk changes aren't merged and released and SPM relies on the Library binaries which don't get published until release, I can't run this test yet. Once the core changes are merged I'll come back an confirm this specific test. All the others run fine

Reply:
functionality removed and only done in core for now